airflow/executors/base_executor.py

```diff
@@ -64,6 +66,20 @@ def __init__(self, parallelism: int = PARALLELISM):
         self.running: Set[TaskInstanceKey] = set()
         self.event_buffer: Dict[TaskInstanceKey, EventBufferValueType] = {}
 
+    @property
+    def job_id(self) -> Optional[str]:
+        """
+        Get job_id for SchedulerJob
+        """
+        return self._job_id
+
+    @job_id.setter
+    def job_id(self, value: str):
+        """
+        Allows SchedulerJob to set job_id variable
+        """
+        self._job_id = value
+
```

Comment:
```suggestion
```

Since we don't do anything special in the getter or setter, lets not bother with them.